src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,221 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars(manufacturer_id, model) VALUES(?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createStatement = connection
+                         .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setLong(1, car.getManufacturer().getId());
+            createStatement.setString(2, car.getModel());
+            createStatement.executeUpdate();
+            ResultSet resultSet = createStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't create car = " + car, throwables);
+        }
+        String createRelationWithCarDrivers = "INSERT INTO cars_drivers(driver_id, car_id) "
+                + "VALUES(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(createRelationWithCarDrivers)) {
+            List<Driver> drivers = car.getDrivers();
+            preparedStatement.setLong(2, car.getId());
+            for (Driver driver : drivers) {
+                preparedStatement.setLong(1, driver.getId());
+                preparedStatement.executeUpdate();
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't add reference between drivers and their cars, "
+                    + "by carId = " + car.getId(), throwables);
+        }
```

Comment:
It's better to create separate method for addition and deletion relationships for `cars_drivers` table since you often use it, rewriting everything.